src/main/drivers/pwm_output.c

```diff
@@ -184,74 +159,120 @@ void pwmCompleteOneshotMotorUpdate(uint8_t motorCount)
         bool overflowed = false;
         // If we have not already overflowed this timer
         for (int j = 0; j < index; j++) {
-            if (motors[j]->tim == motors[index]->tim) {
+            if (motors[j].tim == motors[index].tim) {
                 overflowed = true;
                 break;
             }
         }
         if (!overflowed) {
-            timerForceOverflow(motors[index]->tim);
+            timerForceOverflow(motors[index].tim);
         }
         // Set the compare register to 0, which stops the output pulsing if the timer overflows before the main loop completes again.
         // This compare register will be set to the output value on the next main loop.
-        *motors[index]->ccr = 0;
+        *motors[index].ccr = 0;
     }
 }
 
-void pwmBrushedMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate)
-{
-    const uint32_t hz = PWM_BRUSHED_TIMER_MHZ * 1000000;
-    motors[motorIndex] = pwmOutConfig(timerHardware, PWM_BRUSHED_TIMER_MHZ, hz / motorPwmRate, 0);
-    motors[motorIndex]->pwmWritePtr = pwmWriteBrushed;
-}
-
-void pwmBrushlessMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate, uint16_t idlePulse)
-{
-    const uint32_t hz = PWM_TIMER_MHZ * 1000000;
-    motors[motorIndex] = pwmOutConfig(timerHardware, PWM_TIMER_MHZ, hz / motorPwmRate, idlePulse);
-    motors[motorIndex]->pwmWritePtr = pwmWriteStandard;
-}
-
-void pwmFastPwmMotorConfig(const timerHardware_t *timerHardware, uint8_t motorIndex, uint16_t motorPwmRate, uint16_t idlePulse, uint8_t fastPwmProtocolType)
+void motorInit(motorAndServoConfig_t *motorConfig, uint16_t idlePulse, uint8_t motorCount)
 {
     uint32_t timerMhzCounter;
     pwmWriteFuncPtr pwmWritePtr;
-
-    switch (fastPwmProtocolType) {
-    default:
-    case (PWM_TYPE_ONESHOT125):
-        timerMhzCounter = ONESHOT125_TIMER_MHZ;
-        pwmWritePtr = pwmWriteOneShot125;
-        break;
-    case (PWM_TYPE_ONESHOT42):
-        timerMhzCounter = ONESHOT42_TIMER_MHZ;
-        pwmWritePtr = pwmWriteOneShot42;
-        break;
-    case (PWM_TYPE_MULTISHOT):
-        timerMhzCounter = MULTISHOT_TIMER_MHZ;
-        pwmWritePtr = pwmWriteMultiShot;
-        break;
+        
+    switch (motorConfig->motor_pwm_protocol) {
```

Comment:
Agree, I think my editor reformatted on the closing brace. Will need to check that.